### DISCUSSION TITLE
Error tracing EfficientNet-b0 for c++ use

### DISCUSSION THREAD

**Discussion by bsaunoiss** [Project-MONAI/MONAI#4128] [Q&A] [answered]
Hi all, 

I am working on a project using 3D EfficientNet-b0 to classifiy (32x32x32) patches CT scans. 
Everything works fine concerning the training of the model and the saving of the model. 

My next step is to use the saved model in a C++ script. For that, I am tracing my model :

```
    device = torch.device("cpu")    
    model = monai.networks.nets.EfficientNetBN("efficientnet-b0", in_channels = 1, num_classes = 5, spatial_dims = 3) 
    loss_function = torch.nn.CrossEntropyLoss()
    optimizer = torch.optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
    model.load_state_dict(torch.load(os.path.join(root_dir, "best_metric.pth"), map_location=device))
    model.eval()
    
    # Torch script the model
    scripted = torch.jit.script(model)
    scripted.save(os.path.join(root_dir,"scripted_model.pt"))
    traced = torch.jit.trace(model, torch.rand(1, 1, 32, 32, 32))
    torch.jit.save(traced, os.path.join(root_dir,"traced_model.pt"))
    # Reloaded model to check output
    reloaded_net= torch.jit.load(os.path.join(root_dir,"traced_model.pt"))
    reloaded_net.eval()
```

After the tracing I load it back in my C++ script using :

`torch::jit::load('filename');`

When it comes to this line I have `Unhandled exception at 0x00007FF91814286E (ucrtbase.dll) in App_DeepCollat.exe: Fatal program exit requested.` which doesn't help me a lot ^^

I verified if "traced_model.pt" model was behaving the same way as "best_metric.pth" and everything is normal.
Also, this works perfectly fine when I use other networks such as DenseNet or ResNet but not with EfficientNet. 

What am I missing ? 
Thanks in advance.

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Have you seen this TorchScript related issue before?

Thanks in advance.

**Comment by ericspod** [comment]
I have not seen an error like this before, it's obviously hard to trace where it's coming from being a C++ exception. Would it be possible to test this under some Linux distro? Iy may be a specific Windows error.

**Reply by bsaunoiss** [reply]
Unfortunately I can not (it would take me too much time) but I could run it on MAC if it can help understand if the problem comes from windows or not. What do you think ?

**Reply by ericspod** [reply]
It could be worth trying it on Mac to see if something similar happens but I haven't any more insight on this problem specifically. It could be related to what version of Visual Studio you're using, version of Windows SDK, what patches the OS has, etc. It would take some time to debug without any more idea of root cause.

**Comment by bsaunoiss** [comment, marked as answer]
Quick update for anyone facing this issue ^^
The problem came from the libtorch library that I was using (1.8.1) you need to at least have libtorch 1.9.0 to use the model EfficientNet on C++.
